examples/systemd/rest-server.service

```diff
@@ -10,13 +10,35 @@ Group=www-data
 ExecStart=/usr/local/bin/rest-server --path /path/to/backups
 Restart=always
 RestartSec=5
+# Makes created files group-readable, but inaccessible by others
+UMask=027
 
-# Optional security enhancements
+# Recommended security enhancements using features present in systemd version 247
+CapabilityBoundingSet=
+LockPersonality=true
+MemoryDenyWriteExecute=true
 NoNewPrivileges=yes
 PrivateTmp=yes
+PrivateDevices=true
+PrivateUsers=true
 ProtectSystem=strict
 ProtectHome=yes
+ProtectClock=true
+ProtectControlGroups=true
+ProtectKernelLogs=true
+ProtectKernelModules=true
+ProtectKernelTunables=true
+ProtectProc=invisible
+ProtectHostname=true
+ProcSubset=pid
```

Comment:
I'm wondering whether `ProcSubset` could cause problems, The systemd documentation states the following (https://www.freedesktop.org/software/systemd/man/systemd.exec.html#ProcSubset=):

> Note that Linux exposes various kernel APIs via /proc/, which are made unavailable with this setting. Since these APIs are used frequently this option is useful only in a few, specific cases, and is not suitable for most non-trivial programs.

A quick strace returned accesses to the following two files `/proc/stat` and `/proc/sys/net/core/somaxconn`. The way I understand the `ProcSubset` option these would become inaccessible?

Reply:
Good catch.  I've been using rest-server with `ProcSubset=pid` for a few months, so I suppose those are used for tuning, or adaptation if available.  I've removed the setting.